src/source/ObjcGenerator.scala

```diff
@@ -178,7 +178,7 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
     val self = marshal.typename(objcName, r)
 
     refs.header.add("#import <Foundation/Foundation.h>")
-    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) "../" else "") + marshal.headerName(ident)))
+    refs.body.add("!#import " + q(spec.objcIncludePrefix + (if (r.ext.objc) spec.objcExtendedRecordIncludePrefix else "") + marshal.headerName(ident)))
```

Comment:
Here and elsewhere it looks like there are two different prefixes potentially being added, which would be a problem if both of them were set.  Seems like the code should only add one of them, probably preferring the extended-record prefix if set, and using the other one otherwise.